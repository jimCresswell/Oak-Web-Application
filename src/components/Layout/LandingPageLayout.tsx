import Head from "next/head";
import { FC } from "react";
import styled from "styled-components";
import Link from "next/link";

import SiteHeader from "../SiteHeader";
import SiteFooter from "../SiteFooter";
import { OrganizationJsonLd } from "../../browser-lib/seo/getJsonLd";
import Seo, { SeoProps } from "../../browser-lib/seo/Seo";
import background, { BackgroundProps } from "../../styles/utils/background";
import { OakColorName } from "../../styles/theme";
import footerSections from "../../browser-lib/fixtures/footerSectionLinks";
import Flex from "../Flex";
import IconButtonAsLink from "../Button/IconButtonAsLink";
import { Heading } from "../Typography";

const Container = styled.div<BackgroundProps>`
  display: flex;
  flex-direction: column;
  flex-grow: 1;
  ${background}
`;
const StyledLayout = styled.main`
  display: flex;
  flex-direction: column;
  flex-grow: 1;
  width: 100%;
`;

interface LayoutProps {
  seoProps: SeoProps;
  $background?: OakColorName;
}

/** 1. Titles for SEO should be between 50-60 characters long 
    2. Title should contain app name
    3. SEO descriptions should be between 150-300 characters long */
const LandingPageLayout: FC<LayoutProps> = (props) => {
  const { children, seoProps, $background } = props;
  return (
    <>
      <Seo {...seoProps} />
      <Head>
        <link rel="icon" href="/favicon.ico" />
      </Head>
      <OrganizationJsonLd />
      <Container $background={$background}>
        <SiteHeader />
        <StyledLayout>{children}</StyledLayout>
        <SiteFooter
          footerSections={footerSections}
          footerNotification={
            <Flex justifyContent={"center"} alignItems={"center"}>
              <IconButtonAsLink
                icon={"Rocket"}
                aria-label={"Join new beta oak"}
                href={"/beta/onboarding"}
                size={"tiny"}
                mr={12}
                variant={"primary"}
              />
              <Heading tag="h5" fontSize={16}>
                <Link href={"/beta/onboarding"}>Join Beta</Link>
              </Heading>
            </Flex>
          }
        />
      </Container>
    </>
  );
};

export default LandingPageLayout;
