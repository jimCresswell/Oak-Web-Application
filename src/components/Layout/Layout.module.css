.main {
  display: flex;
  flex-direction: column;
  flex-grow: 1;
  padding: 24px 12px;
  max-width: 1200px;
  width: 100%;
  align-self: center;
}
