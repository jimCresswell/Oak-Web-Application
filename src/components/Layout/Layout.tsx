import Head from "next/head";
import { FC } from "react";
import styled from "styled-components";

import SiteHeader from "../SiteHeader";
import SiteFooter from "../SiteFooter";
import { OrganizationJsonLd } from "../../browser-lib/seo/getJsonLd";
import Seo, { SeoProps } from "../../browser-lib/seo/Seo";

const StyledLayout = styled.main`
  display: flex;
  flex-direction: column;
  flex-grow: 1;
  padding: 24px 12px;
  max-width: 1200px;
  width: 100%;
  align-self: center;
`;

interface LayoutProps {
  seoProps: SeoProps;
}

/** 1. Titles for SEO should be between 50-60 characters long 
    2. Title should contain app name
    3. SEO descriptions should be between 150-300 characters long */
const Layout: FC<LayoutProps> = (props) => {
  const { children, seoProps } = props;
  return (
    <>
      <Seo {...seoProps} />
      <Head>
        <link rel="icon" href="/favicon.ico" />
      </Head>
      <OrganizationJsonLd />
      <SiteHeader />
      <StyledLayout>{children}</StyledLayout>
      <SiteFooter />
    </>
  );
};

export default Layout;
