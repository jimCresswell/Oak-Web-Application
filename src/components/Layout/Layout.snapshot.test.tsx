import renderWithProviders from "../../__tests__/__helpers__/renderWithProviders";

import Layout from "./Layout";

it("renders <Layout> unchanged", () => {
  const { container } = renderWithProviders(
    <Layout
      seoProps={{
        title: "",
        description: "",
        noIndex: undefined,
        image: undefined,
      }}
    />
  );
  expect(container).toMatchSnapshot();
});
