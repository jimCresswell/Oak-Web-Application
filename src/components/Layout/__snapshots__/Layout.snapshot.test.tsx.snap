// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders <Layout> unchanged 1`] = `
<div>
  <header
    class="header"
  >
    <div>
      <span
        class="title"
      >
        Oak
      </span>
      <input
        placeholder="Search"
        type="text"
      />
    </div>
    <button />
  </header>
  <main
    class="main"
  />
  <footer
    class="footer"
  >
    Â© Oak National Academy 
    2022
  </footer>
</div>
`;
